server/utils/validations/loginValidation.js

```diff
@@ -0,0 +1,28 @@
+const yup = require('yup');
+
+const schema = yup.object().shape({
+  preferredContact: yup.string(),
+  email: yup.string().when('preferredContact', {
+    is: 'email',
+    then: yup
+      .string()
+      .email('Please use a valid email address.')
+      .required('Email address is required.'),
+  }),
+  mobileNumber: yup.string().when('preferredContact', {
+    is: 'mobile',
+    then: yup
+      .string()
+      .matches(/^[0-9]{10}$/, 'Mobile number must Contain 10 numbers')
+      .required('Mobile number is required.'),
+  }),
```

Comment:
email and mobile number both should not be required to cause client should be logged by only one of them.

Reply:
I don't think that is a problem because we use.. If the email exists > make it required ..> else do the same with mobile.